Aspose.Slides.WebExtensions/Helpers/ShapeHelper.cs

```diff
@@ -29,10 +29,26 @@ public static string GetShapeAsImageURL<T>(T shape, TemplateContext<T> model)
             }
             else
             {
-                var imgSrcPath = model.Output.GetResourcePath(shape);
-                var slidesPath = model.Global.Get<string>("slidesPath");
-                
-                string result = ShapeHelper.ConvertPathToRelative(imgSrcPath, slidesPath);
+                var imgSrcPath = "";
+                var slidesPath = model.Global.Get<string>("slidesPath");
+
+                try
+                {
+                    imgSrcPath = model.Output.GetResourcePath(shape as Shape);
+                }
+                catch (ArgumentException)
+                {
+                    if (shape is OleObjectFrame frame)
+                    {
+                        imgSrcPath = model.Output.GetResourcePath(frame.SubstitutePictureFormat?.Picture?.Image);

```

Comment:
@polonair ,

Unfotrunatelly, null coalescing needs to be replaced with a more straightforward approach.